### PR TITLE
[p2p] Add P2P monitoring

### PR DESCRIPTION
**Pull request by janet-harmony**

### CODE REVIEW

File: blockchain-watchdog/p2p-monitor.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+func (m *monitor) p2pMonitor(tolerance int, pdServiceKey, chain string, data MetadataContainer) {
+	stdlog.Print("[p2pMonitor] Running p2p connectivity check...")
+	percent := []int{}
+	for _, metadata := range data.Nodes {
+		if metadata.Payload.P2PConnectivity.TotalKnown != 0 {
+			connected := float64(metadata.Payload.P2PConnectivity.Connected)
+			known := float64(metadata.Payload.P2PConnectivity.TotalKnown)
+			connection := int(connected / known * 100)
```
Comment (fxfactorial) [inline blockchain-watchdog/p2p-monitor.go:14]: Possible div by zero

File: blockchain-watchdog/p2p-monitor.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+				}
+			}
+			percent = append(percent, connection)
+		}
+	}
+	sum := 0
+	for _, value := range percent {
+		sum = sum + value
+	}
+	avg := float64(sum) / float64(len(percent))
```
Comment (fxfactorial) [inline blockchain-watchdog/p2p-monitor.go:35]: Div by zero

File: blockchain-watchdog/p2p-monitor.go (hunk 1 above)
Comment (janet-harmony) [reply blockchain-watchdog/p2p-monitor.go:14]: Checked for 0 above

File: blockchain-watchdog/p2p-monitor.go (hunk 2 above)
Comment (janet-harmony) [reply blockchain-watchdog/p2p-monitor.go]: Fixed

**Pull request merged**